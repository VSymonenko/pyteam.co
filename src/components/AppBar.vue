
<template lang="pug">
  transition(name="bar" v-if="show")
    v-toolbar(app :clipped-left="clipped" tabs)
      v-spacer(class="hidden-sm-and-down")
      v-toolbar-title(v-text="title")
      v-spacer
      v-toolbar-items(class="hidden-sm-and-down")
        v-btn(flat v-for="(tab, index) in menu" :key="index" :to="tab.link") {{ tab.name }}
      v-spacer
</template>

<script lang="ts">
import Vue from 'vue';
import Component from 'vue-class-component';

@Component({
  name: 'AppBar',
  props: {
    title: {
      type: String,
    },
    clipped: {
      type: Boolean,
    },
    menu: {
      type: Array,
    },
  },
})

export default class AppBar extends Vue {
  public active: number | null = null;
  private show: boolean = false;

  private mounted(): void {
    setTimeout(() => this.show = true, 500);
  }
}
</script>

<style lang="stylus">
.bar-enter-active, .bar-leave-active {
  transition: opacity .5s;
  transition-delay: 1.5s
}
.bar-enter, .bar-leave-to /* .bar-leave-active  */ {
  opacity: 0;
}
</style>
