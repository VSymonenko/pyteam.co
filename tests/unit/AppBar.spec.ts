import AppBar from '@/components/AppBar.vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import './common';

test('renders correctly', () => {
  const wrapper = shallowMount(AppBar);

  expect(wrapper.isVueInstance()).toBeTruthy();
  expect(wrapper).toMatchSnapshot();
});

describe('AppBar', () => {
  it('renders AppBar.title when passed', () => {
    const title = 'new message';
    const wrapper = shallowMount(AppBar, {
      propsData: { title },
    });
    expect(wrapper.props().title).toContain(title);
  });

  it('renders AppBar.clipped when passed', () => {
    const clipped = true;
    const wrapper = shallowMount(AppBar, {
      propsData: { clipped },
    });
    expect(wrapper.props().clipped).toBe(clipped);
  });

  it('renders a for each item in tabs', () => {
    const menu = {
      about: 'about',
      team: 'team',
    };
    const wrapper = shallowMount(AppBar, {
      propsData: { menu },
      mocks: {
        $t: () => '', // stub i18n
      },
    });
    expect(wrapper.props().menu).toMatchObject(menu);
  });
});
