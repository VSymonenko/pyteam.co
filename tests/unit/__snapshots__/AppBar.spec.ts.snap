// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<vtoolbar-stub scrollthreshold="300" tabs="true" app="app">
  <vspacer-stub class="hidden-sm-and-down"></vspacer-stub>
  <vtoolbartitle-stub></vtoolbartitle-stub>
  <vspacer-stub></vspacer-stub>
  <vtoolbaritems-stub class="hidden-sm-and-down"></vtoolbaritems-stub>
  <vspacer-stub></vspacer-stub>
</vtoolbar-stub>
`;
